pkg/cmd/roachtest/tests/mixed_version_sql_stats.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/mixedversion"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
+	"github.com/cockroachdb/cockroach/pkg/util/httputil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+func registerSqlStatsMixedVersion(r registry.Registry) {
+	// sql-tats/mixed-version tests that requesting sql stats from admin-ui works across
+	// mixed version clusters.
+	r.Add(registry.TestSpec{
+		Name:             "sql-stats/mixed-version",
+		Owner:            registry.OwnerObservability,
+		Cluster:          r.MakeClusterSpec(5),
+		CompatibleClouds: registry.AllClouds,
+		Suites:           registry.Suites(registry.Nightly),
+		Run:              runSQLStatsMixedVersion,
+		Timeout:          15 * time.Minute,
+	})
+}
+
+// runSQLStatsMixedVersion tests that accessing sql stats works across mixed version clusters.
+func runSQLStatsMixedVersion(ctx context.Context, t test.Test, c cluster.Cluster) {
+	roachNodes := c.Range(1, c.Spec().NodeCount-1)
+	workloadNode := c.Node(c.Spec().NodeCount)
+	mvt := mixedversion.NewTest(ctx, t, t.L(), c, roachNodes,
+		mixedversion.MaxUpgrades(2),
+		mixedversion.ClusterSettingOption(
+			install.SecureOption(false),
+		),
+	)
+	flushInterval := 2 * time.Minute
+
+	initWorkload := roachtestutil.NewCommand("./cockroach workload init tpcc").
+		Arg("{pgurl%s}", roachNodes)
+	runWorkload := roachtestutil.NewCommand("./cockroach workload run tpcc").
+		Arg("{pgurl%s}", roachNodes).
+		Option("tolerate-errors")
+
+	setClusterSettings := func(ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+		// SQL Stats are flushed every 10m by default, set it to 3m to speed up the test.
+		l.Printf("setting sql.stats.flush.interval to %s", flushInterval.String())
+		err := h.Exec(r, `SET CLUSTER SETTING sql.stats.flush.interval = $1`, flushInterval.String())
+		return err
+	}
+
+	runGetRequests := func(ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+		// We'll validate that hitting the in-memory tables and either one of the
+		// persisted (full or cached activity) does not return error on any node.
+		// Ideally we'd have prior data in the persisted tables already so that we
+		// can easily query data from the previous aggregation interval. For now
+		// we'll just have to rely on waiting for the flush to occur.
+		expectedTableCount := 2
+
+		// Track the source tables used to service each request.
+		foundStmtTableSources := make(map[string]interface{})
+		foundTxnTableSources := make(map[string]interface{})
```

Comment:
Nit: these sets are generally defined as `map[T]struct{}` to avoid the unnecessary "interface boxing". Shouldn't practically matter in this case.